mail_tracking/models/res_partner.py

```diff
@@ -40,10 +40,15 @@ def email_bounced_set(self, tracking_emails, reason, event=None):
         return partners.write({'email_bounced': True})
 
     def write(self, vals):
-        email = vals.get('email')
-        if email is not None:
-            vals['email'] = email.lower() if email else False
-            vals['email_bounced'] = (
-                bool(email) and
-                self.env['mail.tracking.email'].email_is_bounced(email))
+        if 'email' not in vals:
+            return super(ResPartner, self).write(vals)
+        vals['email'] = vals['email'].lower() if vals['email'] else False
```

Comment:
You shouldn't lower the email here.

Reply:
I agree, but this behaviour is expected by the original code and there's even a test for it.

Reply:
But this is not correct. You must change it.